hs_file_types/models/netcdf.py

```diff
@@ -389,160 +384,97 @@ def check_files_for_aggregation_type(cls, files):
     @classmethod
     def set_file_type(cls, resource, user, file_id=None, folder_path=None):
         """ Creates a NetCDFLogicalFile (aggregation) from a netcdf file (.nc) resource file
-        or a folder """
+        """
 
         log = logging.getLogger()
-        res_file, folder_path = cls._validate_set_file_type_inputs(resource, file_id, folder_path)
-
-        # base file name (no path included)
-        file_name = res_file.file_name
-        # file name without the extension - needed for naming the new aggregation folder
-        nc_file_name = file_name[:-len(res_file.extension)]
-
-        resource_metadata = []
-        file_type_metadata = []
-        upload_folder = ''
-        res_files_to_delete = []
-        # get the file from irods to temp dir
-        temp_file = utils.get_file_from_irods(res_file)
-        temp_dir = os.path.dirname(temp_file)
-
-        # file validation and metadata extraction
-        nc_dataset = nc_utils.get_nc_dataset(temp_file)
-        if isinstance(nc_dataset, netCDF4.Dataset):
-            msg = "NetCDF aggregation. Error when creating aggregation. Error:{}"
-            file_type_success = False
-            # extract the metadata from netcdf file
-            res_dublin_core_meta, res_type_specific_meta = nc_meta.get_nc_meta_dict(temp_file)
-            # populate resource_metadata and file_type_metadata lists with extracted metadata
-            add_metadata_to_list(resource_metadata, res_dublin_core_meta,
-                                 res_type_specific_meta, file_type_metadata, resource)
-
-            # create the ncdump text file
-            dump_file = create_header_info_txt_file(temp_file, nc_file_name)
-            file_folder = res_file.file_folder
-            aggregation_folder_created = False
-            create_new_folder = cls._check_create_aggregation_folder(
-                selected_res_file=res_file, selected_folder=folder_path,
-                aggregation_file_count=1)
-
-            with transaction.atomic():
-                # create a netcdf logical file object to be associated with
-                # resource files
+        with FileTypeContext(aggr_cls=cls, user=user, resource=resource, file_id=file_id,
+                             folder_path=folder_path,
+                             post_aggr_signal=post_add_netcdf_aggregation,
+                             is_temp_file=True) as ft_ctx:
+
+            # base file name (no path included)
+            res_file = ft_ctx.res_file
+            file_name = res_file.file_name
+            # file name without the extension - needed for naming the new aggregation folder
+            nc_file_name = file_name[:-len(res_file.extension)]
```

Comment:
Same doubt as above regarding the trailing dot in nc_file_name

Reply:
Never mind - just realized the extension property on res_file actually includes the dot, so it should be good. This applies to my previous comment regarding the same doubt above.